test/bdd/pkg/connection/connection_sdk_steps.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package connection
+
+import (
+	"fmt"
+
+	"github.com/cucumber/godog"
+	"github.com/google/uuid"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/connection"
+	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/context"
+	connection2 "github.com/hyperledger/aries-framework-go/pkg/store/connection"
+	bddctx "github.com/hyperledger/aries-framework-go/test/bdd/pkg/context"
+	"github.com/hyperledger/aries-framework-go/test/bdd/pkg/didresolver"
+)
+
+// SDKSteps holds connection BDD steps using AFGO's Go SDK.
+type SDKSteps struct {
+	bddContext *bddctx.BDDContext
+}
+
+// NewSDKSteps creates connection SDK steps.
+func NewSDKSteps() *SDKSteps {
+	return &SDKSteps{}
+}
+
+// SetContext is called before every scenario is run with a fresh new bddctx.
+func (s *SDKSteps) SetContext(ctx *bddctx.BDDContext) {
+	s.bddContext = ctx
+}
+
+// RegisterSteps registers the BDD steps on the suite.
+func (s *SDKSteps) RegisterSteps(suite *godog.Suite) {
+	suite.Step(`^"([^"]*)" rotates their connection to "([^"]*)" to new DID$`, s.rotateDID)
+}
+
+func (s *SDKSteps) rotateDID(agentID, otherAgent string) error { // nolint:gocyclo
+	// TODO: find connectionID for connection to other agent (which is given by name, not by DID)
+	// TODO: then call RotateDID.
+	agentCtx := s.bddContext.AgentCtx[agentID]
+
+	myDoc := s.bddContext.PublicDIDDocs[agentID]
+	theirDoc := s.bddContext.PublicDIDDocs[otherAgent]
+
+	didExClient, err := didexchange.New(agentCtx)
+	if err != nil {
+		return err
+	}
+
+	conns, err := didExClient.QueryConnections(&didexchange.QueryConnectionsParams{
```

Comment:
didcommv2 does not require connections, in fact oobv2 is connectionless but target service calls can then create a connection.

maybe as an improvement: add a case of did rotation without a connection? to be discussed if it makes sense..

Reply:
Not sure this works, we still create and save a connection record when we do a didcomm v2 communication, and various places in the code expect a connection record to exist. If we get rid of the connection record, then did rotation will still need some persistence, since it has to add the rotation JWS to outbound messages until it receives a message sent to the new DID (which acknowledges the rotation), and that requires persistent state.